src/main/java/com/iota/iri/service/transactionpruning/DepthPruningCondition.java

```diff
@@ -0,0 +1,36 @@
+package com.iota.iri.service.transactionpruning;
+
+import com.iota.iri.conf.SnapshotConfig;
+import com.iota.iri.controllers.MilestoneViewModel;
+import com.iota.iri.service.snapshot.Snapshot;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.storage.Tangle;
+
+public class DepthPruningCondition implements PruningCondition {
+
+    private final SnapshotConfig config;
+    private final SnapshotProvider snapshotProvider;
+    private final Tangle tangle;
+
+    public DepthPruningCondition(SnapshotConfig config, SnapshotProvider snapshotProvider, Tangle tangle) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Public method and constructor comments are required](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4453821)